camunda-sdk-java/java-common/src/main/java/io/camunda/common/auth/SaaSAuthentication.java

```diff
@@ -53,23 +53,27 @@ public void resetToken(Product product) {
   }
 
   private String retrieveToken(Product product, JwtCredential jwtCredential) {
-    try {
-      HttpPost httpPost = new HttpPost(jwtCredential.getAuthUrl());
-      httpPost.addHeader("Content-Type", "application/json");
-      TokenRequest tokenRequest = new TokenRequest(jwtCredential.getAudience(), jwtCredential.getClientId(), jwtCredential.getClientSecret());
-
-      httpPost.setEntity(new StringEntity(jsonMapper.toJson(tokenRequest)));
-      CloseableHttpClient client = HttpClient.getInstance();
-      CloseableHttpResponse response = client.execute(httpPost);
-      TokenResponse tokenResponse = jsonMapper.fromJson(EntityUtils.toString(response.getEntity()), TokenResponse.class);
-      tokens.put(product, tokenResponse.getAccessToken());
-    } catch (Exception e) {
+      try(CloseableHttpClient client = HttpClient.getInstance()){
+        HttpPost request = buildRequest(jwtCredential);
+        TokenResponse tokenResponse = client.execute(request, response ->
+           jsonMapper.fromJson(EntityUtils.toString(response.getEntity()), TokenResponse.class)
+        );
+        tokens.put(product, tokenResponse.getAccessToken());
+      } catch (Exception e) {
       LOG.error("Authenticating for " + product + " failed due to " + e);
       throw new RuntimeException("Unable to authenticate", e);
     }
     return tokens.get(product);
   }
 
+  private HttpPost buildRequest(JwtCredential jwtCredential) {
+    HttpPost httpPost = new HttpPost(jwtCredential.getAuthUrl());
+    httpPost.addHeader("Content-Type", "application/json");
```

Comment:
nitpick: pretty sure there are constants for the content type, maybe `HttpHeaders.CONTENT_TYPE` and  `MediaType.JSON_UTF_8.toString()`.

Reply:
Yes, I wanted to keep the changes minimal but will tag @1nb0und so he could incorporate your suggestions.